simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/CatalogFragmentBookDetail.kt

```diff
@@ -372,6 +373,47 @@ class CatalogFragmentBookDetail : Fragment() {
     }
 
     this.configureMetadataTable(opds)
+
+    /*
+     * If there's a related feed, enable the "Related books..." item and open the feed
+     * on demand.
+     */
+
+    val feedRelatedOpt = this.parameters.feedEntry.feedEntry.related
+    if (feedRelatedOpt is Some<URI>) {
+      val feedRelated = feedRelatedOpt.get()
+      this.related.setOnClickListener {
+        this.openRelatedFeed(feedRelated)
+      }
+      this.related.isEnabled = true
+    } else {
+      this.related.isEnabled = false
+    }
+  }
+
+  private fun openRelatedFeed(feedRelated: URI) {
+    val context = this.requireContext()
+    val feedModel = this.createOrGetFeedModel(context)
+    val targetFeed =
+      feedModel.resolveFeed(
+        title = context.resources.getString(R.string.catalogRelatedBooks),
+        uri = feedRelated,
+        isSearchResults = false
+      )
+    this.findNavigationController().openFeed(targetFeed)
+  }
+
+  private fun createOrGetFeedModel(
```

Comment:
Minor technical debt here. This method is declared twice over the whole codebase.